al_servicec.js

```diff
@@ -208,21 +261,72 @@ class AzcollectC extends AlServiceC {
             `${checkinValues.awsAccountId}/${checkinValues.region}/${functionName}`, {body: statusBody});
     }
+    
+    _doRegistrationAws(registrationValues) {
+        let statusBody = Object.assign({
+             cf_stack_name : registrationValues.stackName,
+             version : registrationValues.version
+         }, registrationValues.custom_fields);
+         var functionName = encodeURIComponent(registrationValues.functionName);
+         return this.post(`/aws/${registrationValues.collectorType}/` +
+             `${registrationValues.awsAccountId}/${registrationValues.region}/${functionName}`, {body: statusBody});
+     }
 
-    doRegistration(registrationValues) {
-       let statusBody = Object.assign({
-            cf_stack_name : registrationValues.stackName,
-            version : registrationValues.version
-        }, registrationValues.custom_fields);
-        var functionName = encodeURIComponent(registrationValues.functionName);
-        return this.post(`/aws/${registrationValues.collectorType}/` +
-            `${registrationValues.awsAccountId}/${registrationValues.region}/${functionName}`, {body: statusBody});
+     _doDeregistrationAws(registrationValues) {
+         var functionName = encodeURIComponent(registrationValues.functionName);
+         return this.deleteRequest(`/aws/${registrationValues.collectorType}/` +
+             `${registrationValues.awsAccountId}/${registrationValues.region}/${functionName}`);
+     }
+    
+    _doCheckinAzure(checkinValues) {
+        throw '_doCheckinAzure not implemented';
     }
-
-    doDeregistration(registrationValues) {
-        var functionName = encodeURIComponent(registrationValues.functionName);
-        return this.deleteRequest(`/aws/${registrationValues.collectorType}/` +
-            `${registrationValues.awsAccountId}/${registrationValues.region}/${functionName}`);
+    
+    _doRegistrationAzure(registrationValues) {
+        throw '_doRegistrationAzure not implemented';
+    }
+    
+    _doDeregistrationAzure(registrationValues) {
+        throw '_doDeregistrationAzure not implemented';
+    }
+    
+    checkin(checkinValues) {
+        switch(this._collectorType) {
+            case COLLECTOR_TYPES.CWE:
+            case COLLECTOR_TYPES.CWL:
+                return this._doCheckinAws(checkinValues);
+                break;
+            case COLLECTOR_TYPES.O365:
+                return this._doCheckinAzure(checkinValues);
+            default:
+                break;
+        }
+    }
+    
+    register(registrationValues) {
+        switch(this._collectorType) {
+            case COLLECTOR_TYPES.CWE:
+            case COLLECTOR_TYPES.CWL:
+                return this._doRegistrationAws(registrationValues);
+                break;
+            case COLLECTOR_TYPES.O365:
+                return this._doRegistrationAzure(registrationValues);
+            default:
+                break;
```

Comment:
I would suggest to throw an exception for default case in register/deregister/checkin. This should not happen so let's not process it silently

Reply:
This exception will be thrown from the constructor. So the one won't be able to call `register()` for invalid collector type.